src/main/java/de/tum/in/www1/artemis/service/CompetencyService.java

```diff
@@ -62,6 +67,60 @@ public SearchResultPageDTO<Competency> getAllOnPageWithSize(final PageableSearch
         return new SearchResultPageDTO<>(competencyPage.getContent(), competencyPage.getTotalPages());
     }
 
+    /**
+     * Imports all competencies from a course (and optionally their relations) into another.
+     *
+     * @param targetCourse    the course to import into
+     * @param sourceCourse    the course to import from
+     * @param importRelations if competency relations should get imported aswell
+     * @return A list of competencies, each also containing the relations it is the tail competency for.
+     */
+    public List<CompetencyWithTailRelationDTO> importAllCompetenciesFromCourse(Course targetCourse, Course sourceCourse, boolean importRelations) {
+        var competencies = competencyRepository.findAllForCourse(sourceCourse.getId());
+        if (competencies.isEmpty()) {
+            return List.of();
```

Comment:
(nitpick): This is the same functionality but more clearly signals your intend.
```suggestion
            return Collections.emptyList();
```